meteor/sofie-server/business-logic/services/superfly-timeline-builder.ts

```diff
@@ -0,0 +1,435 @@
+			currentPartEnable.duration =
+				activePart.expectedDuration + activePart.getTimings().delayStartOfPiecesDuration
+			shouldAutoNextAtPointInTime = now + currentPartEnable.duration
+		}
+
+		const activeGroup: ActivePartTimelineObjectGroup = {
+			id: `${ACTIVE_GROUP_PREFIX}${activePart.id}`,
+			priority: HIGH_PRIORITY,
+			isGroup: true,
+			children: [],
+			enable: currentPartEnable,
+			layer: '',
+			shouldAutoNextAtPointInTime,
+			content: {
+				type: TimelineObjectType.GROUP,
+				deviceType: DeviceType.ABSTRACT,
+			},
+		}
+
+		activeGroup.children = activePart
+			.getPieces()
+			.flatMap((piece) => this.generateGroupsAndTimelineObjectsForPiece(piece, activePart, activeGroup))
+		return activeGroup
+	}
+
+	private generateGroupsAndTimelineObjectsForPiece(
+		piece: Piece,
+		part: Part,
+		parentGroup: TimelineObjectGroup
+	): TimelineObject[] {
+		const timelineObjectsToReturn: TimelineObject[] = []
+		const pieceEnable: TimelineEnable | undefined = this.createTimelineEnableForPiece(part, piece, parentGroup)
+
+		if (!pieceEnable) {
+			return []
+		}
+
+		if (pieceEnable.start === 'now') {
+			// TODO: We would like to try to not use "now", but they might be necessary for AdLibs. We have to double check when implementing AdLibs.
+			// TODO: This is mainly to alert of if we ever find a Piece with start==="now" while implementing the Timeline. Should be removed when done.
+			throw new UnsupportedOperation(
+				`Found an enable.start="start" for control for Piece: ${piece.id}. We are trying to avoid those if possible.`
+			)
+		}
+
+		const controlForPiece: TimelineObjectControl = this.createTimelineObjectControl(parentGroup, piece, pieceEnable)
+		timelineObjectsToReturn.push(controlForPiece)
+
+		const childGroupForPiece: TimelineObjectGroup = this.createTimelineObjectGroupForPiece(
+			parentGroup,
+			piece,
+			controlForPiece
+		)
+		timelineObjectsToReturn.push(childGroupForPiece)
+
+		if (this.shouldPieceHavePreRollGroup(controlForPiece, piece)) {
+			const preRollControlForPiece: TimelineObjectControl = this.createPreRollGroupForControl(
+				controlForPiece,
+				parentGroup
+			)
+			timelineObjectsToReturn.push(preRollControlForPiece)
+
+			controlForPiece.enable.start = `#${preRollControlForPiece.id} + ${piece.preRollDuration}`
+		}
+
+		// TODO: Core resolves Endcaps, but endCaps seems to be related to when a PieceInstance was stopped which is a concept we don't use.
+		// TODO: EnaCaps seems to be used to check when infinite pieces should stop. That logic should be in Rundown.ts so we shouldn't need to worry about that here
+		// TODO: EndCaps also seems to be used if the Part/Piece needs to stop i.e. a Server will end at some point or an overlay "bundt" is only shown for x seconds. But Blueprint define that on ingest on the TimelineObjects?
+		// TODO: Trying to leave out EndCaps for now and see how the Rundown behaves. Downside? We might have issues with "pieceStartOffset", whatever that is.
```

Comment:
This seems like one todo, masquerading as four. Maybe change it to use a comment block.
I know that it should eventually be removed, but so should the many dozens of todos in the old code also have been.
If you suspect this one is going to stay for a bit, then making it look decent would be nice.

Also i think there is a typo on the second line.

Reply:
Correct, they are technically one comment. The reason for the multiples TODOs is because my editor highlights TODO comments differently, so it's easier for me spot missing ones before committing.

As mentioned, this PR is the current state of building the timeline so this branch doesn't becomes a long lived branched out of touch with the rest of the repo. These TODOs will be gone in the final implementation